### ISSUE TITLE
Exceptions in "Getting Started" colab notebook

### ISSUE THREAD

**Issue by chjort** [poets-ai/elegy#104]
**Describe the bug**
It is not possible to run the "Getting Started" colab notebook as is without getting exceptions.

The notebook is found at https://colab.research.google.com/github/poets-ai/elegy/blob/master/docs/getting-started.ipynb and linked to in the "Getting Started" chapter of the documentation.

When calling ``model.summary`` the following exception occurs:
```python
>>> model.summary(X_train[:64])

---------------------------------------------------------------------------

TypeError                                 Traceback (most recent call last)

<ipython-input-5-fd239b14293e> in <module>()
----> 1 model.summary(X_train[:64])

16 frames

/usr/local/lib/python3.6/dist-packages/jax/core.py in concrete_aval(x)
    829     handler = pytype_aval_mappings.get(typ)
    830     if handler: return handler(x)
--> 831   raise TypeError(f"{type(x)} is not a valid JAX type")
    832 
    833 

TypeError: <class 'elegy.nn.flatten.Flatten'> is not a valid JAX type
```

When calling ``model.fit`` the following exception occurs:
```python
>>> history = model.fit(
......     x=X_train,
......     y=y_train,
......     epochs=100,
......     steps_per_epoch=200,
......     batch_size=64,
......     validation_data=(X_test, y_test),
......     shuffle=True,
......     callbacks=[elegy.callbacks.ModelCheckpoint("model", save_best_only=True)],
...... )

---------------------------------------------------------------------------

TypeError                                 Traceback (most recent call last)

<ipython-input-6-3eecffeab807> in <module>()
      7     validation_data=(X_test, y_test),
      8     shuffle=True,
----> 9     callbacks=[elegy.callbacks.ModelCheckpoint("model", save_best_only=True)],
     10 )

13 frames

/usr/local/lib/python3.6/dist-packages/jax/_src/numpy/lax_numpy.py in _check_arraylike(fun_name, *args)
    296                     if not _arraylike(arg))
    297     msg = "{} requires ndarray or scalar arguments, got {} at position {}."
--> 298     raise TypeError(msg.format(fun_name, type(arg), pos))
    299 
    300 def _check_no_float0s(fun_name, *args):

TypeError: prod requires ndarray or scalar arguments, got <class 'tuple'> at position 0.
```

**Minimal code to reproduce**
The exceptions can also be reproduced by running the following snippet of code in the Colab:
```python
import numpy as np
import jax.numpy as jnp
import jax
import elegy
import optax


class MLP(elegy.Module):
    """Standard LeNet-300-100 MLP network."""
    def __init__(self, n1: int = 300, n2: int = 100, **kwargs):
        super().__init__(**kwargs)
        self.n1 = n1
        self.n2 = n2
    
    def call(self, image: jnp.ndarray) -> jnp.ndarray:
        
        image = image.astype(jnp.float32) / 255.0
        
        mlp = elegy.nn.sequential(
            elegy.nn.Flatten(),
            elegy.nn.Linear(self.n1),
            jax.nn.relu,
            elegy.nn.Linear(self.n2),
            jax.nn.relu,
            elegy.nn.Linear(10),
        )
        
        return mlp(image)


X_train = np.random.uniform(0, 256, size=(60000, 28, 28)).astype(np.uint8)
y_train = np.random.randint(0, 10, size=60000).astype(np.uint8)

X_test = np.random.uniform(0, 256, size=(60000, 28, 28)).astype(np.uint8)
y_test = np.random.randint(0, 10, size=60000).astype(np.uint8)

model = elegy.Model(
    module=MLP(n1=300, n2=100),
    loss=[
        elegy.losses.SparseCategoricalCrossentropy(from_logits=True),
        elegy.regularizers.GlobalL2(l=1e-4),
    ],
    metrics=elegy.metrics.SparseCategoricalAccuracy(),
    optimizer=optax.adam(1e-3),
)


model.summary(X_train[:64])

history = model.fit(
    x=X_train,
    y=y_train,
    epochs=100,
    steps_per_epoch=200,
    batch_size=64,
    validation_data=(X_test, y_test),
    shuffle=True,
    callbacks=[elegy.callbacks.ModelCheckpoint("model", save_best_only=True)],
)
```

**Expected behavior**
The expected output of ``model.fit`` is the summary table:
```
╒═════════════════════╤═══════════════════════╤═════════════════════╤═════════════════╕
│ Layer               │ Outputs Shape         │ Trainable           │   Non-trainable │
│                     │                       │ Parameters          │      Parameters │
╞═════════════════════╪═══════════════════════╪═════════════════════╪═════════════════╡
│ Inputs              │ (64, 28, 28)  uint8   │ 0                   │               0 │
├─────────────────────┼───────────────────────┼─────────────────────┼─────────────────┤
│ flatten   (Flatten) │ (64, 784)     float32 │ 0                   │               0 │
├─────────────────────┼───────────────────────┼─────────────────────┼─────────────────┤
│ linear    (Linear)  │ (64, 300)     float32 │ 235,500    942.0 KB │               0 │
├─────────────────────┼───────────────────────┼─────────────────────┼─────────────────┤
│ relu                │ (64, 300)     float32 │ 0                   │               0 │
├─────────────────────┼───────────────────────┼─────────────────────┼─────────────────┤
│ linear_1  (Linear)  │ (64, 100)     float32 │ 30,100     120.4 KB │               0 │
├─────────────────────┼───────────────────────┼─────────────────────┼─────────────────┤
│ relu_1              │ (64, 100)     float32 │ 0                   │               0 │
├─────────────────────┼───────────────────────┼─────────────────────┼─────────────────┤
│ linear_2  (Linear)  │ (64, 10)      float32 │ 1,010      4.0 KB   │               0 │
├─────────────────────┼───────────────────────┼─────────────────────┼─────────────────┤
│ Outputs   (MLP)     │ (64, 10)      float32 │ 0                   │               0 │
╘═════════════════════╧═══════════════════════╧═════════════════════╧═════════════════╛
Total Parameters:          266,610  1.1 MB
Trainable Parameters:      266,610  1.1 MB
Non-trainable Parameters:  0
```

and the expected output of ``model.fit`` is training progress:
```
...
 
Epoch 99/100
200/200 [==============================] - 1s 4ms/step - l2_regularization_loss: 0.0452 - loss: 0.0662 - sparse_categorical_accuracy: 0.9928 - sparse_categorical_crossentropy_loss: 0.0210 - val_l2_regularization_loss: 0.0451 - val_loss: 0.1259 - val_sparse_categorical_accuracy: 0.9766 - val_sparse_categorical_crossentropy_loss: 0.0808
Epoch 100/100
200/200 [==============================] - 1s 4ms/step - l2_regularization_loss: 0.0450 - loss: 0.0610 - sparse_categorical_accuracy: 0.9953 - sparse_categorical_crossentropy_loss: 0.0161 - val_l2_regularization_loss: 0.0447 - val_loss: 0.1093 - val_sparse_categorical_accuracy: 0.9795 - val_sparse_categorical_crossentropy_loss: 0.0646
```

**Library Info**
Elegy version: 0.2.2
Jax version: 0.2.4
OS info: Colab machine with GPU accelerator

**Comment by cgarciae** [comment]
Hey @chjort, thanks for reporting! 

These issues will be fixed with #92. Newer versions of Jax contain breaking changes (onmistaging) and Colab is probably using the newest version. We currently test with the following versions but we don't force our users to any specific version:

```toml
name = "jaxlib"
version = "0.1.52"

name = "jax"
version = "0.1.75"
```

As a quick fix just change the first code block to:

```
! pip install elegy dataget matplotlib
! pip install jax==0.1.75 jaxlib==0.1.52
```
I've verified this works but it will run on CPU only. @charlielito maybe we should change the getting started tutorial to these version in the meanwhile.

**Comment by chjort** [comment]
Thanks @cgarciae!

Perhaps it could be useful to have the latest tested version of jax and jaxlib under the installation documentation, in case of future breaking changes in either of those. Of course without stating that a specific version is _required_, but just the tested versions.
